src/Elastic.Apm/Report/PayloadSender.cs

```diff
@@ -90,12 +91,12 @@ private async void StartWork()
 					switch (item)
 					{
 						case Payload p:
-							_logger.LogWarning($"Failed sending transaction {p.Transactions.FirstOrDefault()?.Name}");
-							_logger.LogDebug($"{e.GetType().Name}: {e.Message}");
+							_logger.LogWarning(nameof(PayloadSender), "Failed sending transaction {TransactionName}", p.Transactions.FirstOrDefault()?.Name);
```

Comment:
_logger is already scoped so I assume we don't need nameof(PayloadSender) here, correct?

Reply:
Maybe we should define string constants for keys such as "TransactionName" to prevent spelling errors so the line above will become
`_logger.LogWarning($"Failed sending transaction {LogKey.TransactionName}", p.Transactions.FirstOrDefault()?.Name);`
and where `LogKey.TransactionName` value is `{TransactionName}`

Reply:
+1, Ideally will do it in a new PR if that's okay. Rather pull this and #128  in first to keep overview.